docs/source/app_developers_guide/about_event_subscriptions.rst

```diff
@@ -0,0 +1,137 @@
+filter type can specify ``ANY`` (match one or more) or ``ALL`` (must match all).
+The following event filters are supported:
+
+* ``SIMPLE_ANY``:
+   The filter's "match string" must match at least one of the
+   event's attribute values. For example, this filter type with the match string
+   "``abc``" would succeed for an event with the following attributes, because
+   it matches the attribute value ``abc``.
+
+   .. code-block:: none
+
+      Attribute(key="address",value="abc")
+      Attribute(key="address",value="def")
+
+* ``SIMPLE_ALL``:
+   The filter's "match string" must match all of the event's
+   attribute values. This filter type with the match string "``abc``"
+   would fail with the previous example, because it does not match all the
+   attribute values.
+
+* ``REGEX_ANY``:
+   The filter's regular expression must evaluate to a match for
+   one or more of the event's attribute values. For example, this filter type
+   with the match string "``[ad][be][cf]``" would succeed for an event with the
+   following attributes, because it matches the attribute values ``abc`` and
+   ``abbbc``.
+
+   .. code-block:: none
+
+      Attribute(key="address",value="abc")
+      Attribute(key="address",value="abbbc")
+      Attribute(key="address",value="def")
+
+* ``REGEX_ALL``:
+   The filter's regular expression must evaluate to a match for
+   all of the event's attribute values. This filter type with the match string
+   "``[ad][be][cf]``" would fail with the previous example, because it doesn't
+   match all three attribute values. The match string ``[ad][be]*[cf]`` would
+   succeed.
+
+
+Event Subscription Protobuf
+===========================
+
+Event subscriptions are submitted and serviced over a ZMQ socket as messages
+using the protocol buffers format.
+
+An event subscription is represented with the following protobuf messages, which
+are defined in ``sawtooth-core/protos/events.proto``.
+
+.. code-block:: python
+
+    message EventSubscription {
+      string event_type = 1;
+      repeated EventFilter filters = 2;
+    }
+
+    message EventFilter {
+      string key = 1;
+      string match_string = 2;
+
+      enum FilterType {
+          FILTER_TYPE_UNSET = 0;
+          SIMPLE_ANY = 1;
+          SIMPLE_ALL = 2;
+          REGEX_ANY  = 3;
+          REGEX_ALL  = 4;
+        }
+        FilterType filter_type = 3;
+    }
+
+A ``ClientEventsSubscribeRequest`` envelope is used to submit subscription
+request and receive the responses.
+
+.. code-block:: protobuf
+
+  message ClientEventsSubscribeRequest {
+      repeated EventSubscription subscriptions = 1;
+      // The block id (or ids, if trying to walk back a fork) the subscriber last
+      // received events on. It can be set to empty if it has not yet received the
+      // genesis block.
+      repeated string last_known_block_ids = 2;
+  }
+
+The validator responds with a ``ClientEventSubscribeResponse`` message that
```

Comment:
typo "ClientEventsSubscribeResponse" (I know that 's' is terrible but it is more accurate.)